File: module1/module 1 c++/poo_class.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;
typedef unsigned char ui8;
typedef unsigned int ui32;

class Gato
{
public:
    string nombre;
    int edad;
    int peso;
    string color;
};


int main()
{
    Gato tom;
    Gato luna;

    tom.nombre = "Tom";
    tom.edad   = 3;
    tom.peso   = 7;
    tom.color  = "Cafe";

    luna.nombre = "Luna";
    luna.edad   = 2;
    luna.peso   = 5;
    luna.color  = "Gris";

    cout << tom.nombre << endl;
    cout << luna.nombre << endl;

    return 0;
}

File: module1/poo_class.py
# Programación orientada a objetos > Conceptos básicos de POO y UML

class Gato:
    ''' Clase gato simple sin ningún atributo o método '''

if __name__ == '__main__':
    tom = Gato()
    luna = Gato()

    tom.nombre = 'Tom'
    tom.edad = 3
    tom.peso = 7
    tom.color = 'café'

    luna.nombre = 'Luna'
    luna.edad = 2
    luna.peso = 5
    luna.color = 'gris'

    # La siguiente linea muestra la referencia en memoria al objeto
    print(tom)

    # La siguiente linea muestra el valor del atributo asignado al objeto top
    print(tom.nombre)
    print(luna.nombre)
